cmd/pulumi-test-language/tests.go

```diff
@@ -984,6 +984,95 @@ var languageTests = map[string]languageTest{
+	// This test checks how SDKs propagate properties marked as secret to the provider Configure on the gRPC level.
+	"l2-provider-grpc-config-schema-secret": {
+		providers: []plugin.Provider{&providers.ConfigGrpcProvider{}},
+		runs: []testRun{
+			{
+				assert: func(l *L,
+					projectDirectory string, err error,
+					snap *deploy.Snapshot, changes display.ResourceChanges,
+				) {
+					g := &grpcTestContext{l: l, s: snap}
+
+					// Verify the CheckConfig request received by the provider.
+					r := g.CheckConfigReq("config")
+
+					// CheckConfig request gets the secrets in the plain. This is suspect, probably
+					// has to do with secret negotiation happening later in the gRPC provider cycle.
+					assert.Equal(l, "SECRET",
+						r.News.Fields["secretString1"].AsInterface(), "secretString1")
+
+					assertEqualOrJSONEncoded(l, float64(16),
+						r.News.Fields["secretInt1"].AsInterface(), "secretInt1")
+
+					assertEqualOrJSONEncoded(l, float64(123456.7890),
+						r.News.Fields["secretNum1"].AsInterface(), "secretNum1")
+
+					assertEqualOrJSONEncoded(l, true,
+						r.News.Fields["secretBool1"].AsInterface(), "secretBool1")
+
+					assertEqualOrJSONEncoded(l, []any{"SECRET", "SECRET2"},
+						r.News.Fields["listSecretString1"].AsInterface(), "listSecretString1")
+
+					assertEqualOrJSONEncoded(l, map[string]any{"key1": "SECRET", "key2": "SECRET2"},
+						r.News.Fields["mapSecretString1"].AsInterface(), "mapSecretString1")
+
+					// TODO Languages do not agree on the object property casing sent to CheckConfig.
+					//
+					// Node and Go send "secretX".
+					// Python sends "secret_x" though.
+					//
+					// assertEqualOrJSONEncoded(l, map[string]any{"secretX": "SECRET"},
+					// 	r.News.Fields["objSecretString1"].AsInterface(), "objSecretString1")
+
+					// Now verify the Configure request.
+					c := g.ConfigureReq("config")
+
+					// All the fields are coming in as secret-wrapped fields into Configure.
+					assert.Equal(l, secret("SECRET"),
+						c.Args.Fields["secretString1"].AsInterface(), "secretString1")
+
+					assertEqualOrJSONEncodedSecret(l,
+						secret(float64(16)), float64(16),
+						c.Args.Fields["secretInt1"].AsInterface(), "secretInt1")
+
+					assertEqualOrJSONEncodedSecret(l,
+						secret(float64(123456.7890)), float64(123456.7890),
+						c.Args.Fields["secretNum1"].AsInterface(), "secretNum1")
+
+					assertEqualOrJSONEncodedSecret(l,
+						secret(true), true,
+						c.Args.Fields["secretBool1"].AsInterface(), "secretBool1")
+
+					assertEqualOrJSONEncodedSecret(l,
+						secret([]any{"SECRET", "SECRET2"}),
+						[]any{"SECRET", "SECRET2"},
+						c.Args.Fields["listSecretString1"].AsInterface(), "listSecretString1")
+
+					assertEqualOrJSONEncodedSecret(l,
+						secret(map[string]any{"key1": "SECRET", "key2": "SECRET2"}),
+						map[string]any{"key1": "SECRET", "key2": "SECRET2"},
+						c.Args.Fields["mapSecretString1"].AsInterface(), "mapSecretString1")
+
+					// TODO objSecretString1
```

Comment:
Why is this still todo?  We should have an issue attached to all TODOs (by policy in pulumi/pulumi), and probably some more explanation of what's happening here.

Reply:
I believe it's the same problem as above. Python does not agree to other languages. I'll consolidate the todos.